internal/service/project/data_source_projects.go

```diff
@@ -111,7 +111,7 @@ func (d *ProjectsDS) Schema(ctx context.Context, req datasource.SchemaRequest, r
 									"team_id": schema.StringAttribute{
 										Computed: true,
 									},
-									"role_names": schema.ListAttribute{
+									"role_names": schema.SetAttribute{
```

Comment:
as in resource schema

Reply:
I guess no change in UX? For example, can the user still index the role_names?

Reply:
i think so, the state representation is the same

Reply:
What if the users are relying on the order of attributes here?

Reply:
@maastha it's a set in the resource, so I think they don't have currently any guarantee about the order if using the data source.

Reply:
added to the changelog as discussed offline

Reply:
@lantoli I do find it strange to not call this out as[ breaking change](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/changelog-process.md#breaking-changes-and-removals) in the changelog. I believe @EspenAlbert point is valid that users may be accessing a value by index, and with this change that would now error out.
Its definitely a small one, but might be better to be transparent in the changelog.

Reply:
thanks @AgustinBettati and @maastha , let me double check this and will go back to you